src/tracker/tracker.py

```diff
@@ -546,31 +547,34 @@ def compute_known_persons_embeddings(self):
             label_path = os.path.join(self.path_to_known_persons, directory)
             embeddings = []
             for file in os.listdir(label_path):
-                if (
-                    (".jpg" in file.lower())
-                    or (".jpeg" in file.lower())
-                    or (".png" in file.lower())
-                ):
-                    im = Image.open(label_path + "/" + file).convert("RGB")
-                    img_obj = ImageObject(viam_image=None, pil_image=im)
-
-                    detections = self.detector.detect(img_obj)
-
-                    if not detections:
-                        continue
-
-                    # Compute feature vectors for the current detections
-                    batched_features_vectors = self.encoder.compute_features(
-                        img_obj, detections
-                    )
-                    list_of_tensors = list(batched_features_vectors.unbind(dim=0))
-                    embeddings += list_of_tensors
-                else:
+                if not any(file.lower().endswith(suffix)
+                           for suffix in (".jpg", ".jpeg", ".png")):
                     LOGGER.warning(
                         "Ignoring unsupported file type: %s. Only .jpg, .jpeg, and .png files are supported.",  # pylint: disable=line-too-long
                         file,
                     )
+                    continue
+
+                im = Image.open(label_path + "/" + file).convert("RGB")
+                img_obj = ImageObject(viam_image=None, pil_image=im)
+
+                detections = self.detector.detect(img_obj)
 
+                if not detections:
+                    continue
+
+                # Compute feature vectors for the current detections
+                batched_features_vectors = self.encoder.compute_features(
+                    img_obj, detections
+                )
+                list_of_tensors = list(batched_features_vectors.unbind(dim=0))
+                embeddings += list_of_tensors
+
+            if not embeddings:
+                raise NoFacesDetectedError(
+                    f"Unable to recognize any face for supposedly-known "
+                    f"person embedding {directory}!"
+                )
```

Comment:
I think there is an error in the comment, and this is the person embedding, not the face embedding. I say this because the model being run is "detections = self.detector.detect(img_obj)" which I believe is the person detector.

Reply:
So the error should be "unable to recognize any person" rather than "any face"

Could you also fix the comment for the function, which should say "Computes embeddings for known people from the picture directory."

Reply:
Yes, you're totally right! I had misunderstood the distinction between them.